### PR TITLE
Asdsad branch merge

### PR DESCRIPTION
**Pull request by w20k**
@Idrinth could you take a look. Thanks.

### CODE REVIEW

**Comment by w20k** [comment]
Now looks good, what do you think @Idrinth ?

File: mods/chat.js [hunk 1]
```diff
@@ -1,573 +1,603 @@
...
+    replaceInText: function (message, regex, callbacks, lastField) {
+        var complexHandler = function (message, regex, callbacks, lastField) {
+            var partHandler = function (count, callbacks, text, textcontent) {
+                var callbackHandler = function (textcontent, func, text) {
+                    var tmp = func(text);
                     for (var c2 = 0; c2 < tmp.length; c2++) {
-                        if ( tmp[c2] !== undefined ) {
-                            textcontent.push ( tmp[c2] );
+                        if (tmp[c2] === undefined) {
+                            continue;
```
Comment (Idrinth) [inline mods/chat.js]: continue should be replaced by a function with early return statement or a wrapping condition

File: mods/chat.js [hunk 2]
```diff
@@ -1,573 +1,603 @@
...
-                    while ( chat.firstChild ) {
-                        chat.removeChild ( chat.firstChild );
+                if (document.getElementById('idrinth-chat-tab-' + chatId)) {
+                    var chat = document.getElementById('idrinth-chat-tab-' + chatId).getElementsByTagName('ul')[0];
+                    while (chat.firstChild) {
+                        chat.removeChild(chat.firstChild);
                     }
                     for (var userId in idrinth.chat.chatRank[chatId]) {
-                        addMemberElement ( chat, chatId, userId );
+                        if (!idrinth.chat.chatRank[chatId].hasOwnProperty(userId)) continue;
```
Comment (Idrinth) [inline mods/chat.js]: do not skip brackets please, this is too likely to create errors in the future.
Also don't use continues, replace them with a function call with early return

File: mods/chat.js [hunk 3]
```diff
@@ -1,573 +1,603 @@
...
+        var textKey = key || '';
+        var text = {
+            'modify.fail' : 'Can\'t modify that user at the moment',
+            'create.fail' : 'Can\'t create at the moment',
+            'join.fail': 'Can\'t join at the moment',
+            'join.notwork':'Joining didn\'t work',
+            'user.unknown':'The given username for dotd.idrinth.de is unknown, do you want to register it there?',
+            'login.fail':'Login failed in an unexpected way'
+        };
+        return text[textKey];
```
Comment (Idrinth) [inline mods/chat.js]: lacks handling of unknown key and the whole message lookup might be better placed somewhere central, allowing for easierr reuse

File: mods/chat.js (hunk 2 above)
Comment (w20k) [inline mods/chat.js]: It's a must have feature for  "for in object ittiration".
Take a look at docs: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...in 
This paragraph: "Iterating over own properties only"

File: mods/chat.js (hunk 3 above)
Comment (w20k) [inline mods/chat.js]: Thanks, I've made a tickt to move all text msg to a one place. 
It's only a draft version :)

**Comment by Idrinth** [comment]
regarding the whitespace-changes, that should actually be taken care of automatically - just didn't finish debugging the [library](https://github.com/matthiasmullie/minify) that produces invalid code at some places atm.

File: mods/chat.js (hunk 3 above)
Comment (Idrinth) [inline mods/chat.js]: Most important for me is the handling of unknown keys, the moving part can be handled later - it's a good change :)

**Comment by w20k** [comment]
Thanks for your comments ;) I did minor alignment for readability purpose.

File: mods/chat.js (hunk 2 above)
Comment (w20k) [inline mods/chat.js]: Regarding brackets - will keep in mind, just a python|coffeescript habit

File: mods/chat.js [hunk 4]
```diff
@@ -347,7 +346,9 @@ idrinth.chat = {
                         chat.removeChild(chat.firstChild);
                     }
                     for (var userId in idrinth.chat.chatRank[chatId]) {
-                        if (!idrinth.chat.chatRank[chatId].hasOwnProperty(userId)) continue;
+                        if (!idrinth.chat.chatRank[chatId].hasOwnProperty(userId)) {
+                            continue;
```
Comment (Idrinth) [inline mods/chat.js]: That continue here is not really required, would you replace it as well?

Comment (w20k) [inline mods/chat.js]: Removed it for now.

**Comment by Idrinth** [comment]
@w20k didn't mean you needed to remove the check, just turn the condition around and remove the unneeded extra line.

**Comment by Idrinth** [comment]
and thanks for fixing my overlooked dash :+1:

**Comment by w20k** [comment]
Thanks for your comment and points! Was very useful. 
I think we need JS linter to make sure our code is styles are the same in each branch. What IDE you use for JS?

**Pull request merged**